src/topology/config/validation.rs

```diff
@@ -100,7 +100,7 @@ impl Graph {
                     | (Node::Source { ty: ty1 }, Node::Transform { in_ty: ty2, .. })
                     | (Node::Transform { out_ty: ty1, .. }, Node::Transform { in_ty: ty2, .. })
                     | (Node::Transform { out_ty: ty1, .. }, Node::Sink { ty: ty2, .. }) => {
-                        if ty1 != ty2 && ty2 != DataType::Any {
+                        if ty1 != ty2 && ty1 != DataType::Any && ty2 != DataType::Any {
```

Comment:
Did this not require any changes to the tests? I must have missed a case. Could be worth adding a quick one now.

Reply:
Actually, this way caught by a test. cd5ff2954dd52843379563796f143ab11f0e39d3 updates it.

Also, with cd5ff2954dd52843379563796f143ab11f0e39d3 the filtering is skipped if the component supports any event type.